packages/e2e/src/specs/messages/messages-thread-creation.e2e.ts

```diff
@@ -0,0 +1,333 @@
+    const contactPrimaryNumber =
+      await BrowseContactsModal.modalContactPrimaryNumberText
+
+    //TO BE UPDATED AFTER CONTACT NAME HAS DEDICATED DATA-TESTID ATTRIBUTE
+    //await expect(contactName).toHaveText(
+    //existingContactName + " " + existingContactSurname
+    //)
+    await expect(contactPrimaryNumber).toHaveText(existingContactPrimaryNumber)
+  })
+  it("Click available contact and check conversation screen has contact name and phone number displayed above messages", async () => {
+    await waitForClickableAndClick(
+      await BrowseContactsModal.modalContactNameText
+    )
+    const recipientName =
+      await MessagesConversationPage.conversationRecipientNameText
+    const recipientNameText = await recipientName.getText()
+    const recipientNumber =
+      await MessagesConversationPage.conversationRecipientPhoneText
+    const recipientNumberText = await recipientNumber.getText()
+
+    const threadName = await MessagesPage.threadRow
+    await expect(threadName).toHaveText(recipientNameText)
+    await expect(recipientNumberText).toEqual(existingContactPrimaryNumber)
+  })
+
+  it("Insert text to message input, send it and check conversation screen has contact name and phone number displayed above messages", async () => {
+    await MessagesConversationPage.insertTextToMessageInput(
+      "Check conversation title is the same"
+    )
+
+    await waitForClickableAndClick(
+      await MessagesConversationPage.sendMessageButton
+    )
+    const recipientName =
+      await MessagesConversationPage.conversationRecipientNameText
+    const recipientNameText = await recipientName.getText()
+    const recipientNumber =
+      await MessagesConversationPage.conversationRecipientPhoneText
+    const recipientNumberText = await recipientNumber.getText()
+
+    await expect(recipientNameText).toEqual(
+      existingContactName + " " + existingContactSurname
+    )
+    await expect(recipientNumberText).toEqual(existingContactPrimaryNumber)
+  })
+
+  it("click new message and insert part of existing contact name into search contact input", async () => {
+    await waitForClickableAndClick(await MessagesPage.newMessageButton)
+    await MessagesConversationPage.insertTextToSearchContactInput(
+      existingContactName.slice(0, -2)
+    )
+    await expect(
+      MessagesConversationPage.contactSearchResultItem
+    ).toHaveTextContaining(existingContactName)
+    await expect(
+      MessagesConversationPage.contactSearchResultItem
+    ).toHaveTextContaining(existingContactSurname)
+    await expect(
+      MessagesConversationPage.contactSearchResultItem
+    ).toHaveTextContaining(existingContactPrimaryNumber)
+  })
+  it("click New Message and insert part of existing contact number into search contact input", async () => {
+    const recipientInputValue =
+      await MessagesConversationPage.searchContactsInput.getValue()
+    for (let i = 0; i <= (await recipientInputValue).length; i++) {
```

Comment:
Unnecessary await and please, don't use for let i... use forEach method on array :)